### DISCUSSION TITLE
Kamal not running docker application container on server after pulling my latest image. [Help needed]

### DISCUSSION THREAD

**Discussion by arghac14** [basecamp/kamal#1149] [Q&A]
Hi,

I was trying to kamal setup on my server for my application and I noticed while doing kamal setup / deploy the it is not starting my docker application container from the newly built image.

This is the log:

![image](https://github.com/user-attachments/assets/b987474d-a41b-43f0-8f40-493dab5dfcc2)

If you see the log it first runs the kamla-proxy container then it tags the latest image.

After this step, I assume kamal should build and run the image and start the container.

But as the step is missing, it goes to prune the images and prunes the latest image as there is no container running with that image.

Would be of great help if someone knows why this is happening?

Here's how my deploy.yml file format looks currently.

![image](https://github.com/user-attachments/assets/0170f4d8-fdb7-455a-9c3e-8ae06bca4299)

**Comment by maxivak** [comment]
I had the same issue.

remove username in server name:
```
servers:
  workers:
    - <server-ip>
```

instead of
```
servers:
  workers:
    - <ssh-user>@<server-ip>
```

it couldn't find roles for the host named as `<ssh-user>@<server-ip>`.
looks like a bug in kamal where it gets roles for the host
https://github.com/basecamp/kamal/blob/69867e2650051abd3ed4e9d1e1ee7215296f62dd/lib/kamal/commander/specifics.rb#L17-L19

**Reply by giuseb** [reply]
Correct, `ssh-user@` must be removed from the server address.

However, if your `ssh-user` is different from `root`, you need to declare it like so:

```
ssh:
  user: ssh-user
```
